### ISSUE TITLE
performance of the iris_dnn_classifier example

### ISSUE THREAD

**Issue by SergioAlonsoLab** [rstudio/tfestimators#148]
This might be a very basic question.
I'm running the example provided in https://tensorflow.rstudio.com/tfestimators/articles/examples/iris_dnn_classifier.html

``` 
library(tfestimators)

response <- function() "Species"
features <- function() setdiff(names(iris), response())

# split into train, test datasets
set.seed(123)
partitions <- modelr::resample_partition(iris, c(test = 0.2, train = 0.8))
iris_train <- as.data.frame(partitions$train)
iris_test  <- as.data.frame(partitions$test)

# construct feature columns
feature_columns <- feature_columns(
  column_numeric(features())
)

# construct classifier
classifier <- dnn_classifier(
  feature_columns = feature_columns,
  hidden_units = c(10, 20, 10),
  n_classes = 3
)

# construct input function 
iris_input_fn <- function(data) {
  input_fn(data, features = features(), response = response())
}

# train classifier with training dataset
set.seed(123)
train(classifier, input_fn = iris_input_fn(iris_train))

# valuate with test dataset
predictions <- predict(classifier, input_fn = iris_input_fn(iris_test))
evaluation <- evaluate(classifier, input_fn = iris_input_fn(iris_test))
```
To ensure reproducibility, I added a second `set.seed(123)` right before the training.
When comparing the predictions to the actual values in the test dataset, I found that all cases in this dataset have been predicted to belong to the same class:

```
table(iris_test$Species,unlist(predictions$class_ids))
```

Returns:

```            
              2
  setosa     11
  versicolor 10
  virginica   8
```
Is this performance the expected from dnn_classifier?
Compare that to the output of a random forest classification:

```
# random forest

library(randomForest)
set.seed(123)
rforest <- randomForest(Species ~ .,iris_train)
table(iris_test$Species,predict(rforest,newdata=iris_test))
```
Returns a much better classification:
```
             setosa versicolor virginica
  setosa         11          0         0
  versicolor      0         10         0
  virginica       0          1         7
```

Thanks in advance

**Comment by kevinykuo** [comment]
Try training the model for more iterations, e.g.
```
# construct input function 
iris_input_fn <- function(data, epochs = 1) {
  input_fn(data, features = features(), response = response(),
           num_epochs = epochs)
}

# train classifier with training dataset
set.seed(123)
train(classifier, input_fn = iris_input_fn(iris_train, epochs = 50))

# valuate with test dataset
predictions <- predict(classifier, input_fn = iris_input_fn(iris_test))
evaluation <- evaluate(classifier, input_fn = iris_input_fn(iris_test))
table(iris_test$Species,unlist(predictions$class_ids))
#             0  1  2
# setosa     11  0  0
# versicolor  0 10  0
# virginica   0  0  8
```

**Comment by SergioAlonsoLab** [comment]
Thanks a lot, Kevin.
I knew I was missing something obvious...
